src/common/signalk/SKNMEAVisitor.cpp

```diff
@@ -68,12 +73,98 @@ void SKNMEAVisitor::visitSKNavigationAttitude(const SKUpdate &u, const SKPath &p
   sb.setField(8, "ROLL");
 
   _sentences.add(sb.toNMEA() + "\r\n");
+  DEBUG( sb.toNMEA().c_str() );
 };
 
 void SKNMEAVisitor::visitSKNavigationHeadingMagnetic(const SKUpdate &u, const SKPath &p, const SKValue &v) {
-  NMEASentenceBuilder sb2("II", "HDM", 2);
-  sb2.setField(1, SKRadToDeg(v.getNumberValue()), 1);
-  sb2.setField(2, "M");
+  NMEASentenceBuilder sb("II", "HDM", 2);
+  sb.setField(1, SKRadToDeg(v.getNumberValue()), 1);
+  sb.setField(2, "M");
 
-  _sentences.add(sb2.toNMEA() + "\r\n");
+  _sentences.add(sb.toNMEA() + "\r\n");
+  DEBUG( sb.toNMEA().c_str() );
 };
+
+// ***********************  Wind Speed and Angle  ************************
+//  Talker ID: WI Weather Instruments
+//             VW Velocity Sensor, Mechanical?
+//  also seen: IIVWR
+//
+//  NMEA0183  MWV Wind Speed and Angle  --> Deprecated
+//              1  2  3  4
+//              |  |  |  |
+//      $--MWV,x.x,a,x.x,a*hh
+//  1) Wind Angle, 0 to 360 degrees
+//  2) Reference, R = Relative, T = True
+//  3) Wind Speed
+//  4) Wind Speed Units, K/M/N
+//  5) Status, A = Data Valid
+//
+//      VWR Relative Wind Speed and Angle  --> implemented
+//              1  2  3  4  5  6  7  8
+//              |  |  |  |  |  |  |  |
+//      $--VWR,x.x,a,x.x,N,x.x,M,x.x,K*hh
+//  1) Wind direction magnitude in degrees
+//  2) Wind direction Left/Right of bow
+//  3) Speed
+//  4) N = Knots
+//  5) Speed
+//  6) M = Meters Per Second
+//  7) Speed
+//  8) K = Kilometers Per Hour
+// ************************************************************************
+void SKNMEAVisitor::visitSKEnvironmentWindAngleApparent(const SKUpdate &u, const SKPath &p, const SKValue &v) {
+
+  float windAngle;
+  float windSpeed;
+  windAngle = SKRadToDeg(v.getNumberValue());
+
+  // Validation
+  if (windAngle>180||!u.hasEnvironmentWindSpeedApparent()) return;
+
+  windSpeed = u.getEnvironmentWindSpeedTrue();  // in m/s
+  DEBUG("WindSpeed: %f", windSpeed);
+
+  NMEASentenceBuilder sb( "II", "VWR", 8);
```

Comment:
In your other PR you said you would [remove VWR because it is not recommended for new design](https://www.nmea.org/Assets/100108_nmea_0183_sentences_not_recommended_for_new_designs.pdf) and use MWV instead. It looks like what you did is the opposite of what you wanted to do.

Reply:
Yes, you are right!!